### PR TITLE
Added additional filtering for the feed

### PR DESCRIPTION
**Pull request by jmbarne3**
Enhancements:
* Added offset shortcode attribute
* Added default_offset config item
* Added is_important shortcode attribute and config item

### CODE REVIEW

**Comment by jmbarne3** [comment]
The new attributes are used in the `filter` function in the feed class to filter down the results and slice the array appropriately. Unfortunately, I don't think there's a feed option to get just the `is_important` events from the academic calendar, so we have to handle that filtering in the plugin.

File: includes/ucf-academic-calendar-feed.php [hunk 1]
```diff
@@ -20,6 +20,9 @@ public static function get_feed( $args ) {
 
 			$feed_url       = $args['calendar_feed'];
 			$use_cache      = UCF_Acad_Cal_Config::get_option_or_default( 'cache_items' );
+			$offset         = isset( $args['offset'] ) ? intval( $args['offset'] ) : $args['default_offset'];
+			$count          = isset( $args['count'] ) ? intval( $args['count'] ) : $args['default_count'];
+			$isImportant    = $args['is_important'];
```
Comment (cjg89) [inline includes/ucf-academic-calendar-feed.php]: Underscores are preferred over camelCase for PHP variable names (`$is_important`)

File: admin/ucf-academic-calendar-config.php [hunk 2]
```diff
@@ -227,6 +239,24 @@ public static function settings_init() {
...
+
+			add_settings_field(
+				self::$option_prefix . 'is_important',
+				'Pull Important Events',
+				array( 'UCF_Acad_Cal_Config', 'display_settings_field' ),
+				'ucf_acad_cal',
+				'ucf_acad_cal_general',
+				array(
+					'label_for'   => self::$option_prefix . 'is_important',
+					'description' => 'When checked, only events marked as "isImportant" will be pulled from the feed.',
```
Comment (rjucf) [inline admin/ucf-academic-calendar-config.php]: Update isImportant to is_important in the description.

Comment (jmbarne3) [reply admin/ucf-academic-calendar-config.php]: The field in the json is `isImportant`:

```
"category": "Academic Dates and Deadlines",
"url": null,
"summerSession": null,
"isImportant": false,
"uniqueTag": "",
"dtstart": "2017-11-03 00:00:00Z",
"dtend": "",
```

Comment (rjucf) [reply admin/ucf-academic-calendar-config.php]: ah

**Pull request merged**